src/user_input.rs

```diff
@@ -11,32 +11,41 @@ use crate::{
     buttonlike::{MouseMotionDirection, MouseWheelDirection},
 };
 
+/// An [`UserInput::Chord`] could not be created because there were too many input elements
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct ChordCreationError;
```

Comment:
This should probably return the passed in iterator to make error recovery more feasible.

Reply:
Hmm, that does make sense, but at the point of error the iterator will already be consumed for the most part, so I'm not sure how useful it will be for recovery